packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -56,40 +56,56 @@ function getProjectConfiguration(project) {
   return parsedProjectConfiguration;
 }
 
-function _getParsedClassicFilepath(pathSegments, projectConfiguration) {
-  const projectNamePathSeparators = projectConfiguration.pkg.name.split(path.sep);
-
-  pathSegments.splice(
-    0,
-    pathSegments.indexOf(projectNamePathSeparators[0]) + projectNamePathSeparators.length
-  );
-
-  return pathSegments.join(path.sep);
-}
+function _addonFound(filename, projectConfiguration) {
+  const pathSegments = filename.split(path.sep);
+  const addonPaths = projectConfiguration.pkg['ember-addon'].paths;
 
-function _getParsedEmbroiderFilepath(pathSegments) {
-  const RELATIVE_PATH_ROOT = 2;
+  if (!addonPaths) {
+    return;
+  }
 
-  // A typical Embroider pathSegments array would look something like:
-  // ['private', 'var', 'some-hash', 'embroider', 'another-hash', 'tests', 'acceptance', ...]
-  // This strips everything from the first segment up to and including 'another-hash',
-  pathSegments.splice(0, pathSegments.lastIndexOf('embroider') + RELATIVE_PATH_ROOT);
+  const addonNames = addonPaths.map((path) => path.replace('lib/', ''));
 
-  return pathSegments.join(path.sep);
+  return pathSegments.find((addonName) => addonNames.includes(addonName));
 }
 
 /**
  * Get a normalized file path, based on whether the app build is classic or with Embroider
- * @param {object} fileOpts Babel state.file.opts which include root and filename props
+ * @param {object} {root, filename} Babel state.file.opts root & filename
  * @param {object} projectConfiguration Contains project name, ember-addon path info
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
-function getNormalizedFilePath(fileOpts, projectConfiguration) {
-  let { root, filename } = fileOpts;
-  const pathSegments = filename.split(path.sep);
-  const isEmbroider = pathSegments.includes('embroider');
+function getNormalizedFilePath({ root, filename }, projectConfiguration) {
+  const projectPathPrefix = projectConfiguration.pkg.name;
+  const rootDirWithBase = path.join(path.parse(root).dir, path.parse(root).base);
+  const embroiderRegex = new RegExp(`embroider(${path.sep})(.{6})`);
```

Comment:
Can we move this to be a module-scoped constant so we don't have to create a new RegExp for every iteration? I think this pattern should be static for the entire build, correct?